### PR TITLE
Clarify optionality of `databricks_mws_workspaces`.`deployment_name` in docs and examples

### LINKED ISSUE #1314: [ISSUE] Unexpected error provisioning workspace: Deployment name cannot be used until a deployment name prefix is defined
Attempting to create a Databricks workspace using `databricks_mws_workspaces` results in the following unexpected response from the Databricks api: `Deployment name cannot be used until a deployment name prefix is defined. Please contact your Databricks representative`

### Configuration
```hcl
terraform {
  required_providers {
    databricks = {
      source = "databrickslabs/databricks"
      version = "0.5.7"
    }
    aws = {
      source  = "hashicorp/aws"
      version = "4.12.1"
    }
  }
}

provider "aws" {
  region = var.region
}

// initialize provider in "MWS" mode to provision new workspace
provider "databricks" {
  alias    = "mws"
  host     = "https://accounts.cloud.databricks.com"
  username = var.databricks_account_username
  password = var.databricks_account_password
}

resource "databricks_mws_workspaces" "this" {
  provider        = databricks.mws
  account_id      = var.databricks_account_id
  aws_region      = var.region
  workspace_name  = local.prefix
  deployment_name = local.prefix

  credentials_id           = databricks_mws_credentials.this.credentials_id
  storage_configuration_id = databricks_mws_storage_configurations.this.storage_configuration_id
  network_id               = databricks_mws_networks.this.network_id

  token {
    comment = "Terraform"
  }
}
```

### Expected Behavior
A Databricks workspace is provisioned.

### Actual Behavior
```
module.databricks.databricks_mws_workspaces.this: Creating...
╷
│ Error: cannot create mws workspaces: MALFORMED_REQUEST: Failed parameter validation checks: Deployment name cannot be used until a deployment name prefix is defined. Please contact your Databricks representative
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Terraform and provider versions

```
❯ terraform version
Terraform v1.1.9
on darwin_amd64
+ provider registry.terraform.io/databrickslabs/databricks v0.5.7
+ provider registry.terraform.io/hashicorp/aws v4.12.1
+ provider registry.terraform.io/hashicorp/random v3.1.3
```

### Debug Output
```
2022-05-17T11:31:20.395-0400 [WARN]  provider.terraform-provider-databricks_v0.5.7: /api/2.0/accounts/528bd163-c922-4012-8676-e7f79d6250dc/workspaces:400 - MALFORMED_REQUEST: Failed parameter validation checks: Deployment name cannot be used until a deployment name prefix is defined. Please contact your Databricks representative: timestamp=2022-05-17T11:31:20.394-0400
2022-05-17T11:31:20.395-0400 [WARN]  provider.terraform-provider-databricks_v0.5.7: /api/2.0/accounts/528bd163-c922-4012-8676-e7f79d6250dc/workspaces:400 - MALFORMED_REQUEST: Failed parameter validation checks: Deployment name cannot be used until a deployment name prefix is defined. Please contact your Databricks representative: timestamp=2022-05-17T11:31:20.395-0400
```

### Important Factoids

It's not clear to me that this is a provider problem, but if it's truly some sort of a "negotiate this with your Databricks rep" issue, it would be great to call it out in the documentation.

### PR DESCRIPTION
**Pull request by easel**
Fixes #1314

### CODE REVIEW

File: docs/guides/aws-private-link-workspace.md [hunk 1]
```diff
@@ -289,7 +289,8 @@ resource "databricks_mws_workspaces" "this" {
   account_id                 = var.databricks_account_id
   aws_region                 = var.region
   workspace_name             = local.prefix
-  deployment_name            = local.prefix
+  # only set this if you have worked with databricks to set a deployment name
```
Comment (nfx) [inline docs/guides/aws-private-link-workspace.md:292]: probably rename this field from HCL example and add this sentence in the attribute reference in docs/resources/mws_workspaces.md.

File: docs/resources/mws_workspaces.md [hunk 2]
```diff
@@ -206,7 +208,7 @@ The following arguments are available and cannot be changed after workspace is c
 
 * `account_id` - Account Id that could be found in the bottom left corner of [Accounts Console](https://accounts.cloud.databricks.com/).
 * `managed_services_customer_managed_key_id` - (Optional) `customer_managed_key_id` from [customer managed keys](mws_customer_managed_keys.md) with `use_cases` set to `MANAGED_SERVICES`. This is used to encrypt the workspace's notebook and secret data in the control plane.
-* `deployment_name` - (Optional) part of URL: `https://<deployment-name>.cloud.databricks.com`
+* `deployment_name` - (Optional) part of URL: `https://<deployment-name>.cloud.databricks.com`. Must be configured out of band by your databricks rep.
```
Comment (nfx) [inline docs/resources/mws_workspaces.md:211]:
```suggestion
* `deployment_name` - (Optional) part of URL as in `https://<prefix>-<deployment-name>.cloud.databricks.com`. Deployment name cannot be used until a deployment name prefix is defined. Please contact your Databricks representative. Once a new deployment prefix is added/updated, it only will affect the new workspaces created.
```

**Review by nfx — CHANGES_REQUESTED** [review]

File: docs/guides/aws-private-link-workspace.md (hunk 1 above)
Comment (nfx) [inline docs/guides/aws-private-link-workspace.md:292]:
```suggestion
```

File: docs/guides/aws-private-link-workspace.md [hunk 3]
```diff
@@ -289,7 +289,8 @@ resource "databricks_mws_workspaces" "this" {
...
+  # deployment_name            = local.prefix
```
Comment (nfx) [inline docs/guides/aws-private-link-workspace.md:293]:
```suggestion
```

File: docs/resources/mws_private_access_settings.md [hunk 4]
```diff
@@ -32,7 +32,8 @@ resource "databricks_mws_workspaces" "this" {
   account_id                 = var.databricks_account_id
   aws_region                 = var.region
   workspace_name             = local.prefix
-  deployment_name            = local.prefix
+  # only set this if you have worked with databricks to set a deployment name
```
Comment (nfx) [inline docs/resources/mws_private_access_settings.md:35]:
```suggestion
```

File: docs/resources/mws_private_access_settings.md [hunk 5]
```diff
@@ -32,7 +32,8 @@ resource "databricks_mws_workspaces" "this" {
...
+  # deployment_name            = local.prefix
```
Comment (nfx) [inline docs/resources/mws_private_access_settings.md:36]:
```suggestion
```

File: docs/resources/mws_workspaces.md [hunk 6]
```diff
@@ -68,7 +68,8 @@ resource "databricks_mws_workspaces" "this" {
   provider        = databricks.mws
   account_id      = var.databricks_account_id
   workspace_name  = var.prefix
-  deployment_name = var.prefix
+  # only set this if you have worked with databricks to set a deployment name
+  # deployment_name = var.prefix
```
Comment (nfx) [inline docs/resources/mws_workspaces.md:72]:
```suggestion
```

**Review by nfx — APPROVED** [review]

**Comment by easel** [comment]
Thanks for the merge!

**Pull request merged**